### PR TITLE
Shelley get reward account balance

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-302 / #1750 

### Overview

Adds implementation of network layer `getAccountBalance` for shelley.

### Comments

The integration test is not passing because stake pools are not connected with the test cluster network.

### CODE REVIEW

**Review by paweljakubas — APPROVED** [review]
great job! just make the test pending before we can run it, also change `Delegations` type alias a little bit using the type aliases from cardano-api (as we use them below):
```
type ShelleyCredentialStaking            = Shelley.Credential Shelley.Staking Shelley.TPraosStandardCrypto
type ShelleyVerificationKeyHashStakePool = Shelley.KeyHash Shelley.StakePool Shelley.TPraosStandardCrypto
```
(https://github.com/input-output-hk/cardano-node/blob/master/cardano-api/src/Cardano/Api/Types.hs#L144 and https://github.com/input-output-hk/cardano-node/blob/master/cardano-api/src/Cardano/Api/Types.hs#L153)

**Review by Anviking — APPROVED** [review]
lgtm

File: lib/shelley/src/Cardano/Wallet/Shelley/Network.hs [hunk 1]
```diff
@@ -206,43 +218,57 @@ withNetworkLayer
...
+        connectNodeTipClient handlers
+
+    queryRewardQ <- connectDelegationRewardsClient handlers
 
     action
         NetworkLayer
             { currentNodeTip = liftIO $ _currentNodeTip nodeTipVar
             , nextBlocks = _nextBlocks
-            , initCursor = _initCursor
+            , initCursor = _initCursor queryRewardQ
```
Comment (Anviking) [inline lib/shelley/src/Cardano/Wallet/Shelley/Network.hs:238]: Wait, why is `queryRewardQ` both read from by the shared `connectDelegationRewardsClient` and each wallet client? That seems odd. Shouldn't `queryRewardQ` _only_ be handled by the shared client?

_trying removing it on my branch_

Comment (rvl) [reply lib/shelley/src/Cardano/Wallet/Shelley/Network.hs:238]: Oops. I meant to remove that.

**Pull request merged**